library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerCommands.java

```diff
@@ -8,35 +8,119 @@
+     */
     void addDownloadBatchCallback(DownloadBatchStatusCallback downloadBatchCallback);
 
+    /**
+     * Removes a {@link DownloadBatchStatusCallback} from the internal list of callbacks,
+     * removing notifications on {@link DownloadBatchStatus} changes.
+     *
+     * @param downloadBatchCallback to remove from the internal list of callbacks.
+     */
     void removeDownloadBatchCallback(DownloadBatchStatusCallback downloadBatchCallback);
 
+    /**
+     * Retrieves all stored batches from the persistence layer and queues them ready for download,
+     * repopulating the internal map.
+     *
+     * @param callback to be notified when all downloads are queued.
+     */
     void submitAllStoredDownloads(AllStoredDownloadsSubmittedCallback callback);
 
+    /**
+     * Retrieves a list of {@link DownloadBatchStatus} synchronously. Clients should
+     * specify their own Threading mechanism.
+     *
+     * @return a list of {@link DownloadBatchStatus}.
+     */
     @WorkerThread
     List<DownloadBatchStatus> getAllDownloadBatchStatuses();
 
+    /**
+     * Retrieves a list of {@link DownloadBatchStatus} passing to the {@link AllBatchStatusesCallback}.
+     *
+     * @param callback to pass the list of {@link DownloadBatchStatus} to.
+     */
     void getAllDownloadBatchStatuses(AllBatchStatusesCallback callback);
 
+    /**
+     * Retrieves {@link DownloadFileStatus} with the associated {@link DownloadBatchId} and {@link DownloadFileId}.
+     * Will return null if a {@link DownloadFileStatus} is not found with the matching {@link DownloadBatchId} and {@link DownloadFileId}.
+     * This call is synchronous, clients should specify their own Threading mechanism.
+     *
+     * @param downloadBatchId of the {@link DownloadFileStatus} to retrieve.
+     * @param downloadFileId  of the {@link DownloadFileStatus} to retrieve.
+     * @return {@link DownloadFileStatus} or null.
+     */
     @Nullable
     @WorkerThread
     DownloadFileStatus getDownloadFileStatusWithMatching(DownloadBatchId downloadBatchId, DownloadFileId downloadFileId);
 
+    /**
+     * Retrieves {@link DownloadFileStatus} with the associated {@link DownloadBatchId} and {@link DownloadFileId}.
+     * Will return null if a {@link DownloadFileStatus} is not found with the matching {@link DownloadBatchId} and {@link DownloadFileId}.
+     *
+     * @param downloadBatchId of the {@link DownloadFileStatus} to retrieve.
+     * @param downloadFileId  of the {@link DownloadFileStatus} to retrieve.
+     * @param callback        to pass the {@link DownloadFileStatus} to.
+     */
     void getDownloadFileStatusWithMatching(DownloadBatchId downloadBatchId, DownloadFileId downloadFileId, DownloadFileStatusCallback callback);
 
+    /**
+     * Changes the {@link ConnectionType} that must be satisfied in order to download.
+     * Will pause all current downloads that are violating this condition.
```

Comment:
I was confused about this because I couldn't see where it resumes once the condition is satisfied. It doesn't seem to schedule a job or anything to check.